core/network/subnet.go

```diff
@@ -295,7 +295,7 @@ func FindSubnetIDsForAvailabilityZone(zoneName string, subnetsToZones map[Id][]s
 	matchingSubnetIDs := set.NewStrings()
 	for subnetID, zones := range subnetsToZones {
 		zonesSet := set.NewStrings(zones...)
-		if zonesSet.Contains(zoneName) {
+		if zonesSet.Size() == 0 && zoneName == "" || zonesSet.Contains(zoneName) {
```

Comment:
I am concerned with the special casing we have introduced here, particularly given that we do not call this with an empty zone name (see `subnetsForZone` in the Azure provider).